Firebase.h

```diff
@@ -25,49 +25,97 @@
 #include <WiFiClientSecure.h>
 #include <ESP8266HTTPClient.h>
 
-// FirebaseError represents a Firebase API error with a code and a
-// message.
-class FirebaseError {
- public:
-  operator bool() const { return _code < 0; }
-  int code() const { return _code; }
-  const String& message() const { return _message; }
-  void reset() { set(0, ""); }
-  void set(int code, const String& message) {
-    _code = code;
-    _message = message;
-  }
- private:
-  int _code = 0;
-  String _message = "";
-};
+//TODO(edcoyne) split these into multiple files.
+
+class FirebaseCall;
+class FirebaseEventStream;
 
-// Firebase is the connection to firebase.
+// Primary client to the Firebase backend.
 class Firebase {
  public:
   Firebase(const String& host);
   Firebase& auth(const String& auth);
-  const FirebaseError& error() const {
-    return _error;
+
+  // Fetch result at "path" to a local variable. If the value is too large you will exceed
+  // local memory.
+  FirebaseCall get(const String& path);
```

Comment:
> we can never clear the response from the buffer because we don't know if you will ask for it 3 hours later

I agree with this generally, but given that the interaction between the arduino sketch and the lib is not likely to be concurrent (1 request at a time for a given Firebase object). Do you still see this as an issue if we hold the buffer until the next request (and overwrite it)?.

The esp8266 http lib is currently modeled that way:
https://github.com/esp8266/Arduino/blob/0834ec62a1d7ac832026c33f6c6b97e1c76ebba8/libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.h#L199 with a bunch of mutable state within the http client object that get reset on every request.